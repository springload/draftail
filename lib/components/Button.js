import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../components/Icon';

const onMouseDown = (onClick, e) => {
    e.preventDefault();

    onClick();
};

const Button = ({ name, active, label, title, icon, onClick }) =>
    <button
        name={name}
        className={`RichEditor-styleButton${active
            ? ' RichEditor-activeButton'
            : ''}`}
        type="button"
        title={title}
        onMouseDown={onMouseDown.bind(null, onClick)}
    >
        {icon ? <Icon name={icon} /> : null}
        {label}
    </button>;

Button.propTypes = {
    name: PropTypes.string,
    active: PropTypes.bool,
    label: PropTypes.string,
    title: PropTypes.string,
    icon: PropTypes.string,
    onClick: PropTypes.func,
};

Button.defaultProps = {
    name: null,
    active: false,
    label: null,
    title: null,
    icon: null,
    onClick: () => {},
};

export default Button;
