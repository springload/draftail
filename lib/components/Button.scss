.toolbar-button {
    display: inline-block;
    margin-top: $controls-spacing;
    padding: $button-spacing;
    border-radius: 0;
    background: transparent;
    color: inherit;
    border: 1px solid transparent;
    font-size: 1rem;
    font-weight: 600;
    cursor: pointer;

    &:hover,
    &:focus {
        color: inherit;
        background-color: rgba(0, 0, 0, 0.05);
    }

    &:focus {
        border: 1px solid $color-editor-chrome-active;
        outline: 0;
    }

    &--active {
        background-color: rgba(0, 0, 0, 0.1);
        border: 1px solid rgba(0, 0, 0, 0.2);

        &:hover,
        &:focus {
            background-color: rgba(0, 0, 0, 0.1);
        }
    }
}
