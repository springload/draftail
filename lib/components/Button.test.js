import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button';

describe('Button', () => {
    it('exists', () => {
        expect(Button).toBeDefined();
    });

    it('empty', () => {
        expect(shallow(<Button />)).toMatchSnapshot();
    });

    it('#name', () => {
        expect(shallow(<Button name="TEST" />)).toMatchSnapshot();
    });

    it('#label', () => {
        expect(shallow(<Button label="Test label" />)).toMatchSnapshot();
    });

    it('#icon', () => {
        expect(shallow(<Button icon="icon-test" />)).toMatchSnapshot();
    });

    it('#active', () => {
        expect(shallow(<Button active={true} />)).toMatchSnapshot();
    });

    it('#onClick', () => {
        const onClick = jest.fn();
        const event = { preventDefault: jest.fn() };
        shallow(<Button onClick={onClick} />).simulate('mousedown', event);
        expect(onClick).toHaveBeenCalledTimes(1);
        expect(event.preventDefault).toHaveBeenCalledTimes(1);
    });
});
