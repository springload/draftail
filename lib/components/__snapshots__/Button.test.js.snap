// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button #active 1`] = `
<button
  className="RichEditor-styleButton RichEditor-activeButton"
  name={null}
  onMouseDown={[Function]}
  title={null}
  type="button"
/>
`;

exports[`Button #icon 1`] = `
<button
  className="RichEditor-styleButton"
  name={null}
  onMouseDown={[Function]}
  title={null}
  type="button"
>
  <Icon
    className=""
    name="icon-test"
    title={null}
  />
</button>
`;

exports[`Button #label 1`] = `
<button
  className="RichEditor-styleButton"
  name={null}
  onMouseDown={[Function]}
  title={null}
  type="button"
>
  Test label
</button>
`;

exports[`Button #name 1`] = `
<button
  className="RichEditor-styleButton"
  name="TEST"
  onMouseDown={[Function]}
  title={null}
  type="button"
/>
`;

exports[`Button empty 1`] = `
<button
  className="RichEditor-styleButton"
  name={null}
  onMouseDown={[Function]}
  title={null}
  type="button"
/>
`;
